source/val/validation_state.cpp

```diff
@@ -73,6 +73,9 @@ bool IsInstructionInLayoutSection(ModuleLayoutSection layout, SpvOp op) {
         case SpvOpGroupDecorate:
         case SpvOpGroupMemberDecorate:
         case SpvOpDecorationGroup:
+        case SpvOpDecorateId:
```

Comment:
Hmm. Maybe this should delegate to IsAnnotationInst.  :-)
Not requested for this CL.